Shared/Samples/Edit feature attachments/EditFeatureAttachmentsView.swift

```diff
@@ -0,0 +1,215 @@
+                    model.featureLayer.clearSelection()
+                    do {
+                        let result = try await mapProxy.identify(
+                            on: model.featureLayer,
+                            screenPoint: screenPoint,
+                            tolerance: 5
+                        )
+                        guard let features = result.geoElements as? [ArcGISFeature],
+                              let feature = features.first else {
+                            return
+                        }
+                        try await model.setSelectedFeature(for: feature)
+                        if let location = mapProxy.location(fromScreenPoint: screenPoint) {
+                            model.updateCalloutPlacement(to: location)
+                        }
+                    } catch {
+                        self.error = error
+                    }
+                }
+        }
+        .errorAlert(presentingError: $error)
+    }
+}
+
+private extension EditFeatureAttachmentsView {
+    // MARK: - AttachmentSheetView
+    
+    struct AttachmentSheetView: View {
+        /// The error shown in the error alert.
+        @State private var error: Error?
+        /// The data model for the sample.
+        @ObservedObject var model: Model
+        
+        var body: some View {
+            Form {
+                Section {
+                    List {
+                        ForEach($model.attachments.indices, id: \.self) { index in
+                            let attachment = model.attachments[index]
+                            AttachmentView(attachment: attachment, onDelete: { attachment in
+                                Task {
+                                    do {
+                                        try await model.deleteAttachment(attachment)
+                                    } catch {
+                                        self.error = error
+                                    }
+                                }
+                            })
+                        }
+                    }
+                }
+                Section {
+                    AddAttachmentView(onAdd: {
+                        Task {
+                            do {
+                                guard let pngData = UIImage(
+                                    named: "PinBlueStar"
+                                )?.pngData() else {
+                                    return
+                                }
+                                try await model.addAttachment(
+                                    named: "Attachment",
+                                    type: "png",
+                                    dataElement: pngData
+                                )
+                            } catch {
+                                self.error = error
+                            }
+                        }
+                    })
+                }
+            }
```

Comment:
```suggestion
        /// The action to dismiss the sheet.
        @Environment(\.dismiss) private var dismiss: DismissAction
        
        var body: some View {
            NavigationStack {
                Form {
                    Section {
                        List {
                            ForEach($model.attachments.indices, id: \.self) { index in
                                let attachment = model.attachments[index]
                                AttachmentView(attachment: attachment, onDelete: { attachment in
                                    Task {
                                        do {
                                            try await model.deleteAttachment(attachment)
                                        } catch {
                                            self.error = error
                                        }
                                    }
                                })
                            }
                        }
                    }
                    Section {
                        AddAttachmentView(onAdd: {
                            Task {
                                do {
                                    guard let pngData = UIImage(
                                        named: "PinBlueStar"
                                    )?.pngData() else {
                                        return
                                    }
                                    try await model.addAttachment(
                                        named: "Attachment",
                                        type: "png",
                                        dataElement: pngData
                                    )
                                } catch {
                                    self.error = error
                                }
                            }
                        })
                    }
                }
                .navigationTitle("Attachments")
                .navigationBarTitleDisplayMode(.inline)
                .toolbar {
                    ToolbarItem(placement: .confirmationAction) {
                        Button("Done") {
                            dismiss()
                        }
                    }
                }
```
I have a few suggestions for the sheet design so it is consistent with other samples:
- Use a Dismiss action and "Done" button to dismiss the sheet
- Embed the view in a navigation stack and add a navigation title to help clarify the purpose of the sheet

Reply:
![screen4](https://github.com/Esri/arcgis-maps-sdk-swift-samples/assets/13937686/0442ab67-208d-419e-abed-8bf9ef30d250)